server/controllers/user-books-controller.js

```diff
@@ -0,0 +1,351 @@
+                    });
+                  } else {
+                    Books
+                      .findOne({
+                        where: {
+                          id: bookId,
+                          isActive: true,
+                        },
+                      })
+                      .then((foundBook) => {
+                        if (foundBook) {
+                          if (foundBook.bookQuantity < 1) {
+                            // if all copies of book have been borrowed
+                            res.status(200).json({
+                              status: 'Unsuccessful',
+                              message: 'Book Unavailable',
+                            });
+                          } else {
+                            BookLendings
+                              .create({
+                                bookId: foundBook.id,
+                                userId: userLoginDetails.id,
+                                borrowDate: borrowdate,
+                                dueDate: duedate,
+                              })
+                              .then((crtLending) => {
+                                foundBook
+                                  .update({
+                                    bookQuantity: foundBook.bookQuantity - 1,
+                                  })
+                                  .then((bookBorrowed) => {
+                                    res.status(202).json({
+                                      status: 'Success',
+                                      message: 'Book Successfully Borrowed',
+                                      lendId: crtLending.id,
+                                      book: bookBorrowed.bookName,
+                                      QuantityLeft: bookBorrowed.bookQuantity,
+                                      dueDate: crtLending.dataValues.dueDate,
+                                    });
+                                  })
+                                  .catch(error => res.status(501).send(error));
+                              })
+                              .catch(error => res.status(501).send(error));
+                          }
+                        } else {
+                          res.status(400).json({
+                            status: 'Unsuccessful',
+                            message: 'Invalid Book',
+                          });
+                        }
+                      })
+                      .catch(error => res.status(501).send(error));
+                  }
+                })
+                .catch(error => res.status(501).send(error));
+            }
+          } else {
+            res.status(501).json({
+              status: 'Unsuccessful',
+              message: 'Try again!??',
+            });
+          }
+        })
+        .catch(error =>
+          res.status(404).json({
+            status: 'Unsuccessful',
+            message: error,
```

Comment:
more descriptive error messages
Abstract re-useable code block to a function and call the function in your controller